src/ucp/core/ucp_request.inl

```diff
@@ -662,6 +662,18 @@ ucp_request_recv_data_unpack(ucp_request_t *req, const void *data,
     }
 }
 
+static UCS_F_ALWAYS_INLINE ucp_recv_desc_t*
+ucp_recv_desc_get(ucp_worker_h worker, size_t length)
+{
+    /* TODO: Optimize */
+    uint32_t rp  = ucs_roundup_pow2(length + 1); /* make sure length is not 0 */
+    unsigned idx = ucs_popcount(worker->am_mps_map & (rp - 1));
+
+    ucs_assertv(length < UINT_MAX, "length %zu", length);
+
```

Comment:
this is a fast path, need to optimize
something like:
`&worker->am_mps[64 - builtin_clz(length)]`
to avoid potentially expensive popcount, am_mps could contain pointers to mpools and several entries would point to same mpool